### DISCUSSION TITLE
How to extract tracks from queue?

### DISCUSSION THREAD

**Discussion by danyaridiger** [Androz2091/discord-player#1660] [Q&A] [answered]
I need to extract information (duration, name, etc.) about tracks from queue using `useQueue` utility or some other way. But `tracks` property of queue has turned into custom `Map` object in v6. How can i work with queue tracks for getting information?

**Comment by twlite** [comment, marked as answer]
Hello @danyaridiger, the tracks object in v6 is the internal Queue object (https://discord-player.js.org/docs/classes/@discord-player/utils/Queue) which provides methods such as `.toArray()` to turn the queue into array of tracks, `.at()` to get the track at the given index and so on. Please refer to the link I provided in this message for all of the available methods.

**Reply by danyaridiger** [reply]
Queue created by `player.nodes.create` method also has `initializer` property?

**Reply by twlite** [reply]
Yes, whatever you'd pass on to `nodeOptions` while using the play method goes there.

**Reply by danyaridiger** [reply]
Thanks!
